Maple2.Server.Game/Manager/ClubManager.cs

```diff
@@ -0,0 +1,222 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Google.Protobuf.Collections;
+using Maple2.Database.Storage;
+using Maple2.Model.Enum;
+using Maple2.Model.Error;
+using Maple2.Model.Game;
+using Maple2.Model.Game.Club;
+using Maple2.Server.Core.Sync;
+using Maple2.Server.Game.Packets;
+using Maple2.Server.Game.Session;
+using Maple2.Server.Game.Util.Sync;
+using Maple2.Tools.Extensions;
+using Serilog;
+
+namespace Maple2.Server.Game.Manager;
+
+public class ClubManager : IDisposable {
+    private readonly GameSession session;
+
+    public Club Club { get; init; }
+    public long Id => Club.Id;
+
+    private readonly CancellationTokenSource tokenSource;
+
+    private readonly ILogger logger = Log.Logger.ForContext<ClubManager>();
+
+    private ClubManager(GameSession session, Club club) {
+        this.session = session;
+        tokenSource = new CancellationTokenSource();
+        Club = club;
+    }
+
+    public static ClubManager? Create(ClubInfo clubInfo, GameSession session) {
+        Club? club = SetClub(session, clubInfo);
+        if (club == null) {
+            Log.Error("Failed to set club for {Session}", session);
+            return null;
+        }
+
+        return new ClubManager(session, club);
+    }
+
+    public void Dispose() {
+        session.Dispose();
+        tokenSource.Dispose();
+
+        foreach (ClubMember member in Club.Members.Values) {
+            member.Dispose();
+            if (member.CharacterId == session.CharacterId) {
+                using GameStorage.Request db = session.GameStorage.Context();
+                db.SaveClubMember(member);
```

Comment:
WorldServer should be source of truth. Is there session specific info that we would need to be saving here?

Reply:
Ah this was to update their log out time on ClubMember

Reply:
Is it possible to just get that information on `WorldServer` vs `UpdatePlayer`?

Reply:
I suppose just adding 
`        member.LoginTime = info.UpdateTime > 0 ? info.UpdateTime : member.LoginTime;` (same as how it's in GuildMember and ClubMember atm)

In `SyncUpdate` in the Club/GuildManager would do it?

Reply:
`SyncUpdate` is in `GameServer`, isn't the problem with saving the time to db which means we need the update to be set in `WorldServer`?

I'm also thinking we might not even want to denormalize `LoginTime` like this. Might be better to just add it to `PlayerInfo` so it's consistent everywhere? `Character` db could store `LoginTime` in a normalized fashion.